### PR TITLE
Feature/singlepagebugfix

### PR DESCRIPTION
**Pull request by jnnklhmnn**
- enhaced tests for javascript link
- solved issue with clicked links and pushed meta keys https://github.com/moccu/picnic/issues/21

### CODE REVIEW

File: src/singlepage/views/Singlepage.js [hunk 1]
```diff
@@ -37,10 +37,21 @@ class View extends BaseView {
...
+			(event.button && event.button === 1)
+		) {
+
+			return;
+		}
 		// Test if protocol, hostname, port is equal to the current location...
 		// Test if link opens as target="_self"...
 		// Test if link is no download...
-		if (
+		else if (
```
Comment (schorfES) [inline src/singlepage/views/Singlepage.js]: `else if` not required here. Get back to the previous version and please add one more space before the comments block.

File: tests/singlepage/views/singlepage.js [hunk 2]
```diff
@@ -14,13 +14,19 @@ function __setLocation(url) {
 	window.history.replaceState({}, undefined, url);
 }
 
-function __triggerClickEvent(self) {
+function __triggerClickEvent(self ,options) {
```
Comment (schorfES) [inline tests/singlepage/views/singlepage.js]: Swap space and comma ;-)

File: tests/singlepage/views/singlepage.js [hunk 3]
```diff
@@ -14,13 +14,19 @@ function __setLocation(url) {
...
 	var isDefaultPrevented = false;
 	self.view.$el.on('click', 'a', function(event) {
 		isDefaultPrevented = event.isDefaultPrevented();
 		event.preventDefault();
 	});
-	self.view.$el.find('a').trigger(new $.Event('click'));
+
```
Comment (schorfES) [inline tests/singlepage/views/singlepage.js:23]: You don't need to separate here. It doesn't matter if the second parameter is defined or undefined for the options parameter of the `$.Event` constructor.

File: tests/singlepage/views/singlepage.js [hunk 4]
```diff
@@ -199,6 +205,74 @@ QUnit.test(
...
+		// hier dem event das cmd key mitgeben.
+		isDefaultPrevented =  __triggerClickEvent(this, options);
+
+		assert.ok(!(isDefaultPrevented), 'it does not prevent default behaviour');
+		assert.ok(callback.notCalled, 'it does not fire a callback');
+	}
+);
+
+QUnit.test(
+	'should open a new tab when ctrl metakey is pressed and link clicked',
```
Comment (schorfES) [inline tests/singlepage/views/singlepage.js]: ctrl is not a "metakey"

File: tests/singlepage/views/singlepage.js [hunk 5]
```diff
@@ -199,6 +205,74 @@ QUnit.test(
...
+
+		isDefaultPrevented =  __triggerClickEvent(this, options);
+
+		assert.ok(!(isDefaultPrevented), 'it does not prevent default behaviour');
+		assert.ok(callback.notCalled, 'it does not fire a callback');
+	}
+);
+
+QUnit.test(
+	'should open a new tab when shift metakey is pressed and link clicked',
```
Comment (schorfES) [inline tests/singlepage/views/singlepage.js]: shift is not a "metakey"

**Comment by jnnklhmnn** [comment]
issues have been reworked on, check again or merge if fine

**Pull request merged**